src/Function/Bundles/Related.agda

```diff
@@ -0,0 +1,290 @@
+------------------------------------------------------------------------
+
+-- The Agda standard library
+--
+-- Relatedness for the function hierarchy
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Function.Bundles.Related where
+
+open import Level
+open import Relation.Binary using (Sym; Reflexive; Trans; IsEquivalence; Setoid; IsPreorder; Preorder)
+import Function.Related as R
+open import Function.Bundles
+open import Function.Base
+open import Function.Equality using (_⟨$⟩_)
+import Function.Equivalence as Eq
+import Function.Injection   as Inj
+import Function.Inverse     as Inv
+import Function.LeftInverse as LeftInv
+import Function.Surjection  as Surj
+import Relation.Binary.PropositionalEquality as P
+
+open import Function.Properties.Inverse
+import Function.Construct.Symmetry as Symmetry
+import Function.Construct.Identity as Identity
+import Function.Construct.Composition as Composition
+
+open import Data.Product
+
+infix 4 _∼[_]_
+
+_∼[_]_ : ∀ {ℓ₁ ℓ₂} → Set ℓ₁ → R.Kind → Set ℓ₂ → Set _
+A ∼[ R.implication         ] B = A ⟶ B
+A ∼[ R.reverse-implication ] B = B ⟶ A
+A ∼[ R.equivalence         ] B = A ⇔ B
+A ∼[ R.injection           ] B = A ↣ B
+A ∼[ R.reverse-injection   ] B = B ↣ A
+A ∼[ R.left-inverse        ] B = A ↪ B
+A ∼[ R.surjection          ] B = A ↠ B
+A ∼[ R.bijection           ] B = A ⤖ B
+
+-- A non-infix synonym.
+
+Related : R.Kind → ∀ {ℓ₁ ℓ₂} → Set ℓ₁ → Set ℓ₂ → Set _
+Related k A B = A ∼[ k ] B
+
+module _ {ℓ₁ ℓ₂} {A : Set ℓ₁} {B : Set ℓ₂} where
```

Comment:
This would be a bit annoying, as we pattern-match on the `Kind` here, and because it's an implicit argument we'd have to add four `{_}`'s to every branch. The anonymous module seems like a cleaner solution to me.

Reply:
You'll be happy to learn you can name anonymous arguments to avoid exactly this problem, e.g. `toRelated {K = R.implication}` :smile:

Reply:
I think I remember trying this in the past in other situations and in didn't work, but it could have been that I didn't use spaces... Thanks!